roles/overpass-api/tasks/compile.yml

```diff
@@ -0,0 +1,68 @@
+- name: install packages
+  apt:
+    pkg: "{{ item }}"
+  with_items:
+    - autoconf
+    - automake1.11
+    - expat
+    - git
+    - g++
+    - libtool
+    - libexpat1-dev
+    - make
+    - zlib1g-dev
+    - bzip2
+    - wget
+    - liblz4-1
+    - liblz4-dev
+
+- name: download {{ overpass_release }} release of overpass-api
+  unarchive:
+    src: "https://github.com/drolbr/Overpass-API/archive/osm3s-{{ overpass_release }}.tar.gz"
+    dest: "{{ project_dir }}/"
+    validate_certs: false
+    remote_src: true
+  become: yes
+  become_user: "{{ user }}"
+
+- name: compile overpass-api
+  shell: |
+    autoscan &&
+    aclocal-1.11 &&
+    autoheader &&
+    libtoolize &&
+    automake-1.11 --add-missing &&
+    autoconf &&
+    ./configure --enable-lz4 CXXFLAGS="-O2" &&
+    make -j $(nproc --all)
+  register: make
+  args:
+    chdir: "{{ project_dir }}/Overpass-API-osm3s-{{ overpass_release }}/src"
+  become: yes
+  become_user: "{{ user }}"
+
+- name: install overpass-api executables
+  shell: make install
+  args:
+    chdir: "{{ project_dir }}/Overpass-API-osm3s-{{ overpass_release }}/src"
+
+- name: install overpass-api cgi-bin
+  command: cp -r {{ project_dir }}/Overpass-API-osm3s-{{ overpass_release }}/src/cgi-bin/ {{ webroot_dir }}/
+
+- name: remove packages
+  apt:
+    pkg: "{{ item }}"
+    state: absent
+  with_items:
+    - autoconf
+    - automake1.11
+    - expat
+    - git
+    - g++
+    - libtool
+    - libexpat1-dev
+    - make
+    - zlib1g-dev
+    - bzip2
+    - wget
+    - liblz4-dev
```

Comment:
I don't really like to remove packages - it could cause issue if they are needed for something else. For exemple, bzip2, wget, git and make are useful independently of this ansible script.